### PR TITLE
Feat/pipeline dashboard

### PR DESCRIPTION
**Pull request by Joewizy**

### CODE REVIEW

File: .github/workflows/dashboard_app_pylint.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+name: Dashboard App
+
+on: [push, pull_request]
```
Comment (djeck1432) [inline .github/workflows/dashboard_app_pylint.yml:3]: leave just `push` please

File: apps/dashboard_app/charts/main_chart_figure.py [hunk 2]
```diff
@@ -103,7 +121,7 @@ def get_main_chart_figure(
     customdata = get_custom_data(data)
 
     # Add bars for each protocol and the total liquidable debt
-    for col in color_map_protocol.keys():
```
Comment (djeck1432) [inline apps/dashboard_app/charts/main_chart_figure.py:106]: why did you change `keys-> items`?

File: apps/dashboard_app/helpers/ekubo.py [hunk 3]
```diff
@@ -81,14 +90,13 @@ def fetch_liquidity(self, bids: bool = True) -> dict[str, str | list[float]]:
         :return: dict[str, str | list[float]]
         """
 
-        params = self.params_for_bids
```
Comment (djeck1432) [inline apps/dashboard_app/helpers/ekubo.py:84]: why diid you changes this code?

File: apps/dashboard_app/helpers/ekubo.py [hunk 4]
```diff
@@ -81,14 +90,13 @@ def fetch_liquidity(self, bids: bool = True) -> dict[str, str | list[float]]:
...
+        params = self.params_for_bids if bids else self.params_for_asks
         if not bids:
-            params = self.params_for_asks
```
Comment (djeck1432) [inline apps/dashboard_app/helpers/ekubo.py:86]: why did you remove this code?

File: apps/dashboard_app/helpers/ekubo.py [hunk 5]
```diff
@@ -81,14 +90,13 @@ def fetch_liquidity(self, bids: bool = True) -> dict[str, str | list[float]]:
...
             logging.warning(
                 "Using collateral token as base token and debt token as quote token."
             )
 
-        response = requests.get(self.URL, params=params)
+        response = requests.get(self.URL, params=params, timeout=10)
```
Comment (djeck1432) [inline apps/dashboard_app/helpers/ekubo.py:99]: why did you add this code?

File: apps/dashboard_app/helpers/ekubo.py [hunk 6]
```diff
@@ -99,13 +107,12 @@ def fetch_liquidity(self, bids: bool = True) -> dict[str, str | list[float]]:
                 data["prices"], data["quantities"] = zip(*liquidity[data["type"]])
             except ValueError:
                 time.sleep(300 if bids else 5)
-                self.fetch_liquidity(bids=True)
```
Comment (djeck1432) [inline apps/dashboard_app/helpers/ekubo.py:102]: why did you remove this code

File: apps/dashboard_app/helpers/loans_table.py [hunk 7]
```diff
@@ -8,9 +11,25 @@
 
 
 def get_protocol(state: State) -> str:
+    """
```
Comment (djeck1432) [inline apps/dashboard_app/helpers/loans_table.py:14]: please, sync with master, this code is outdated

File: apps/dashboard_app/helpers/protocol_stats.py [hunk 8]
```diff
@@ -119,11 +126,16 @@ def get_supply_stats(
...
+    Get collateral stats for the dashboard.
+    :param states: States zklend, nostra_alpha, nostra_mainnet
+    :return: DataFrame with collateral stats
+    """
     data = []
     for state in states:
         protocol = get_protocol(state=state)
         token_collaterals = defaultdict(float)
-        for token in TOKEN_SETTINGS:
+        for token in TOKEN_SETTINGS.items():
```
Comment (djeck1432) [inline apps/dashboard_app/helpers/protocol_stats.py:138]: why did you change this code?

File: apps/dashboard_app/helpers/protocol_stats.py [hunk 9]
```diff
@@ -172,11 +184,16 @@ def get_collateral_stats(
...
+    Get debts for the dashboard.
+    :param states: States zklend, nostra_alpha, nostra_mainnet
+    :return: DataFrame with debt stats
+    """
     data = []
     for state in states:
         protocol = get_protocol(state=state)
         token_debts = defaultdict(float)
-        for token in TOKEN_SETTINGS:
+        for token in TOKEN_SETTINGS.items():
```
Comment (djeck1432) [inline apps/dashboard_app/helpers/protocol_stats.py:196]: why did you change this code?

File: apps/dashboard_app/helpers/tools.py [hunk 10]
```diff
@@ -106,33 +118,37 @@ def get_prices(token_decimals: dict[str, int]) -> dict[str, float]:
...
+                 ,token, decimals, token_info.get('decimals'))
             continue
 
         prices[token] = token_info.get("currentPrice")
 
     return prices
 
 
-def add_leading_zeros(hash: str) -> str:
+def add_leading_zeros(address: str) -> str:
```
Comment (djeck1432) [inline apps/dashboard_app/helpers/tools.py:135]: why did you change this code

**Review by djeck1432 — CHANGES_REQUESTED** [review]
SOrry, but I think you don't manage this task, you can't manage copy/paste from chatgpt  even

File: apps/dashboard_app/helpers/ekubo.py (hunk 5 above)
Comment (Joewizy) [reply apps/dashboard_app/helpers/ekubo.py:99]: Hello there every code I modified was based on suggestion from pylint like error: C0206: Consider iterating with .items()(consider using-duct-items) or warning W3101 about not adding the timeout Argus in the request.get().

**Review by Joewizy — COMMENTED** [review]
removed "pull request"

**Comment by Joewizy** [comment]
@djeck1432 synced

**Comment by djeck1432** [comment]
Screenshot 2024-11-29 at 18 07 44
Screenshot 2024-11-29 at 18 07 53
If I can see those, that means you didn't sync at all

**Comment by Joewizy** [comment]
@djeck1432 sorry I forgot to commit changes everything has been synced now

File: apps/dashboard_app/helpers/loans_table.py [hunk 11]
```diff
@@ -8,9 +11,25 @@
 
 
 def get_protocol(state: State) -> str:
```
Comment (djeck1432) [inline apps/dashboard_app/helpers/loans_table.py:13]: This code is old and should be removed, please here is the master's branch code
https://github.com/CarmineOptions/derisk-research/blob/master/apps/dashboard_app/helpers/loans_table.py#L10

File: apps/dashboard_app/helpers/protocol_stats.py [hunk 12]
```diff
@@ -68,7 +75,7 @@ def get_supply_stats(
     for state in states:
         protocol = get_protocol(state=state)
         token_supplies = {}
-        for token in TOKEN_SETTINGS:
+        for token in TOKEN_SETTINGS.items():
```
Comment (djeck1432) [inline apps/dashboard_app/helpers/protocol_stats.py:78]: remove `items()`

File: apps/dashboard_app/helpers/protocol_stats.py [hunk 13]
```diff
@@ -119,11 +126,16 @@ def get_supply_stats(
...
+    Get collateral stats for the dashboard.
+    :param states: States zklend, nostra_alpha, nostra_mainnet
+    :return: DataFrame with collateral stats
+    """
     data = []
     for state in states:
         protocol = get_protocol(state=state)
         token_collaterals = defaultdict(float)
-        for token in TOKEN_SETTINGS:
+        for token in TOKEN_SETTINGS():
```
Comment (djeck1432) [inline apps/dashboard_app/helpers/protocol_stats.py:138]: remove `()`

File: apps/data_handler/README.md [hunk 14]
```diff
@@ -95,6 +95,31 @@ Go to bash
 docker-compose exec backend bash
 ```
 
+
+## Running Migration Command:
```
Comment (djeck1432) [inline apps/data_handler/README.md:99]: Please fix that, look as it's in master branch
https://github.com/CarmineOptions/derisk-research/blob/master/apps/data_handler/README.md

**Review by djeck1432 — CHANGES_REQUESTED** [review]
Still see a few code places where you changed the code, which you shouldn't change
also bad rebasing with master
Please, you have 12 hours to fixt all issues, otherwise you will be unassigned

**Comment by Joewizy** [comment]
@djeck1432 I have finally resolved all issues,  and updated my code with the master branch please confirm

**Review by djeck1432 — APPROVED** [review]
Looks good to me
